assets/wizards/newspack/views/settings/connections/custom-events.tsx

```diff
@@ -0,0 +1,143 @@
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { WizardError } from '../../../../errors';
+import { useWizardApiFetch } from '../../../../hooks/use-wizard-api-fetch';
+import { Button, Grid, Notice, TextControl } from '../../../../../components/src';
+
+/**
+ * Validate GA4 Measurement ID.
+ *
+ * @see   https://measureschool.com/ga4-measurement-id/
+ *
+ * @param measurementId Measurement ID to validate
+ * @return boolean      True if the measurement ID is valid, false otherwise
+ */
+function isValidGA4MeasurementID( measurementId = '' ) {
+	const ga4Pattern = /^G-[A-Za-z0-9]{10,}$/;
+	return ga4Pattern.test( measurementId );
+}
+
+/**
+ * Analytics Custom Events screen.
+ */
+function CustomEvents() {
+	const [ ga4Credentials, setGa4Credentials ] = useState< Ga4Credentials >(
+		window.newspackSettings.connections.sections.customEvents
+	);
+	const { wizardApiFetch, errorMessage, resetError, setError } = useWizardApiFetch(
+		'newspack-settings/connections/custom-events'
+	);
+
+	useEffect( () => {
+		if ( ! isValidGA4MeasurementID( ga4Credentials.measurement_id ) ) {
+			setError(
+				new WizardError(
+					__(
+						'You need a valid Measurement ID (e.g. "G-ABCDE12345") to activate Newspack Custom Events.',
+						'newspack-plugin'
+					),
+					'ga4_invalid_measurement_id'
+				)
+			);
+			return;
+		}
+		if ( ! ga4Credentials.measurement_protocol_secret ) {
+			setError(
+				new WizardError(
+					__(
+						'You need a valid Measurement API Secret to activate Newspack Custom Events.',
+						'newspack-plugin'
+					),
+					'ga4_invalid_measurement_protocol_secret'
+				)
+			);
+			return;
+		}
+		resetError();
+	}, [ ga4Credentials.measurement_id, ga4Credentials.measurement_protocol_secret ] );
+
+	function updateGa4Credentials() {
+		wizardApiFetch< Ga4Credentials >(
+			{
+				path: '/newspack/v2/wizard/analytics/ga4-credentials',
+				method: 'POST',
+				data: {
+					measurement_id: ga4Credentials.measurement_id,
+					measurement_protocol_secret: ga4Credentials.measurement_protocol_secret,
+				},
+			},
+			{
+				onSuccess( fetchedData ) {
+					window.newspackSettings.connections.sections.customEvents = {
+						...window.newspackSettings.connections.sections.customEvents,
+						...fetchedData,
+					};
+					setGa4Credentials( fetchedData );
+				},
+			}
+		);
+	}
+
+	return (
+		<div className="newspack__analytics-configuration">
+			<div className="newspack__analytics-configuration__header">
+				<p>
+					{ __(
+						"Newspack already sends some custom event data to your GA account, but adding the credentials below enables enhanced events that are fired from your site's backend. For example, when a donation is confirmed or when a user successfully subscribes to a newsletter.",
+						'newspack-plugin'
+					) }
+				</p>
+			</div>
+
+			{ errorMessage && <Notice isError noticeText={ errorMessage } /> }
```

Comment:
Nitpick: Rendering the error message above the form fields results in the form moving around as the measurement ID field is being modified. I would suggest placing the error below the form inputs instead to avoid this behavior

Reply:
This was from existing UI. I have moved the error notice to below the inputs as you've suggested.